src/exojax/spec/opachord.py

```diff
@@ -22,8 +22,8 @@ def chord_geometric_matrix_lower(height, radius_lower):
         
     """
     radius_upper = radius_lower + height
-    fac_left = jnp.sqrt(radius_upper[None, :]**2 - radius_lower[:, None]**2)
-    fac_right = jnp.sqrt(radius_lower[None, :]**2 - radius_lower[:, None]**2)
+    fac_left = jnp.sqrt(jnp.tril(radius_upper[None, :]**2 - radius_lower[:, None]**2))
+    fac_right = jnp.sqrt(jnp.tril(radius_lower[None, :]**2 - radius_lower[:, None]**2, k=-1))
     raw_matrix = 2.0 * (fac_left - fac_right) / height
     return jnp.tril(raw_matrix)
```

Comment:
Do you still need this `jnp.tril(raw_matrix)`?

Reply:
No, I don't. I will remove it.